### PR TITLE
Delete duplicate files - hipify-perl and helper scripts

### PR DESCRIPTION
**Pull request by aakanksha555**
SWDEV-309659 - [ROCm QA][IV][QR][G]: hipifyclang package installation fails in the latest builds

hipify-clang package now installs hipify-perl and related helper scripts along with it. https://github.com/ROCm-Developer-Tools/HIPIFY/pull/445

### CODE REVIEW

**Review by emankov — CHANGES_REQUESTED** [review]
There are a lot of places in the HIP repo where hipify-perl and helper scripts are mentioned: documentation, samples, and it looks like all of them have to be cleaned up.
We should get to know what are the new places to refer instead.

**Comment by aakanksha555** [comment]
> There are a lot of places in the HIP repo where hipify-perl and helper scripts are mentioned: documentation, samples, and it looks like all of them have to be cleaned up. We should get to know what are the new places to refer instead.

@emankov could you help with any other references I may have missed? I updated link to hipconvertinplace-perl.sh.
hipify package also installs to opt/rocm/hip/bin the references to $HIP_DIR/bin would stay the same?

File: docs/markdown/hip_porting_guide.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ and provides practical suggestions on how to port CUDA code and work through com
 - Starting the port on a CUDA machine is often the easiest approach, since you can incrementally port pieces of the code to HIP while leaving the rest in CUDA. (Recall that on CUDA machines HIP is just a thin layer over CUDA, so the two code types can interoperate on nvcc platforms.) Also, the HIP port can be compared with the original CUDA code for function and performance.
 - Once the CUDA code is ported to HIP and is running on the CUDA machine, compile the HIP code using the HIP compiler on an AMD machine.
 - HIP ports can replace CUDA versions: HIP can deliver the same performance as a native CUDA implementation, with the benefit of portability to both Nvidia and AMD architectures as well as a path to future C++ standard support. You can handle platform-specific features through conditional compilation or by adding them to the open-source HIP infrastructure.
-- Use **[bin/hipconvertinplace-perl.sh](https://github.com/ROCm-Developer-Tools/HIP/blob/master/bin/hipconvertinplace-perl.sh)** to hipify all code files in the CUDA source directory.
```
Comment (emankov) [inline docs/markdown/hip_porting_guide.md:59]: remove "bin/", please

File: docs/markdown/hip_porting_guide.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ and provides practical suggestions on how to port CUDA code and work through com
...
+- Use **[bin/hipconvertinplace-perl.sh](https://github.com/ROCm-Developer-Tools/HIPIFY/blob/master/bin/hipconvertinplace-perl.sh)** to hipify all code files in the CUDA source directory.
 
 ### Scanning existing CUDA code to scope the porting effort
 The hipexamine-perl.sh tool will scan a source directory to determine which files contain CUDA code and how much of that code can be automatically hipified.
```
Comment (emankov) [inline docs/markdown/hip_porting_guide.md]: Add a hyperlink to hipexamine-perl.sh in HIPIFY repo as well, please.

**Comment by emankov** [comment]
1. Remove from [Readme.md](https://github.com/ROCm-Developer-Tools/HIP/blob/develop/README.md):
    * **hipify-perl** : Script based tool to convert CUDA code to portable CPP. Converts CUDA APIs and kernel builtins.
    * **hipexamine-perl.sh** : Script to scan the directory, find all code, and report statistics on how much can be ported with HIP (and identify likely features not yet supported).
    * **hipconvertinplace-perl.sh** : Script to scan the directory, find all code, and convert the found CUDA code to HIP reporting all unconverted things.
2. I'd remove [hipify-cmakefile](https://github.com/ROCm-Developer-Tools/HIP/blob/develop/bin/hipify-cmakefile) as well not to mislead. It is a very old, outdated, and incorrect script for hipification. 
3. [README.md](https://github.com/ROCm-Developer-Tools/HIP/blob/develop/samples/0_Intro/square/README.md) in square sample: rename all `/home/user` to `/opt/rocm`.
4. Remove the file https://github.com/ROCm-Developer-Tools/HIP/blob/develop/tests/hipify-clang/unit_tests/libraries/cuSPARSE/cuSPARSE_12.cu

**Comment by emankov** [comment]
> @emankov could you help with any other references I may have missed? I updated link to hipconvertinplace-perl.sh.
> hipify package also installs to opt/rocm/hip/bin the references to $HIP_DIR/bin would stay the same?

`/opt/rocm/`, `$HIP_DIR`, and `$HIP_PATH` are ok; no changes are needed as long as the HIPIFY tools package is being installed into the same directory as HIP itself.

**Comment by aakanksha555** [comment]
> > @emankov could you help with any other references I may have missed? I updated link to hipconvertinplace-perl.sh.
> > hipify package also installs to opt/rocm/hip/bin the references to $HIP_DIR/bin would stay the same?
> 
> `/opt/rocm/`, `$HIP_DIR`, and `$HIP_PATH` are ok; no changes are needed as long as the HIPIFY tools package is being installed into the same directory as HIP itself.

Yep, hipify-clang package installs to /opt/rocm/hip - https://github.com/ROCm-Developer-Tools/HIPIFY/blob/master/CMakeLists.txt#L126

**Review by emankov — DISMISSED** [review]
LGTM, but let's wait for the review from Maneesh.

+ Please, make the rest of the requested changes in hip_porting_guide.md.
Thanks!

File: docs/markdown/hip_porting_guide.md (hunk 1 above)
Comment (aakanksha555) [reply docs/markdown/hip_porting_guide.md:59]: Done.

File: docs/markdown/hip_porting_guide.md (hunk 2 above)
Comment (aakanksha555) [reply docs/markdown/hip_porting_guide.md]: Done.

**Comment by aakanksha555** [comment]
> LGTM, but let's wait for the review from Maneesh.
> 
> * Please, make the rest of the requested changes in hip_porting_guide.md.
>   Thanks!

Sure thing. Thanks!

**Review by emankov — APPROVED** [review]

**Review by mangupta — APPROVED** [review]
LGTM

**Comment by aakanksha555** [comment]
Thanks, could either of you merge in the changes, I don't have the perms.

**Pull request merged**